test/cuda.jl

```diff
@@ -26,6 +26,17 @@ end
   @test g_gpu |> collect ≈ g
 end
 
+@testset "sum(f, x)" begin
+  a = Float32.(-4:4)
```

Comment:
gradient of abs at 0 is ... contentious 
```suggestion
  a = Float32.([-1.5, -9.0, 2.4, -1.3, 0.01])
```